### PR TITLE
[MIG] web_widget_float_formula: Migrate to 10.0

### PR DESCRIPTION
**Pull request by kellylougheed**

### CODE REVIEW

**Review by lasley — APPROVED** [review]
Thanks @kellylougheed 

Not sure why CLA bot is spazzing out, Oleg has been covered under LasLabs ECLA for a while. I think it actually meant to say that Kelly's wasn't in, and just got confused on rebase or something. We should have her's too now, under SMDrugstore ECLA.

**Review by lasley — CHANGES_REQUESTED** [review]
Now that Runbot has rebuilt though, I can see that this doesn't work functionally. Please double check.

**Comment by kellylougheed** [comment]
Ready for review @lasley

File: web_widget_float_formula/static/src/js/web_widget_float_formula.js [hunk 1]
```diff
@@ -52,12 +53,7 @@ odoo.define('web_widget_float_formula', function(require) {
             },
 
             _process_formula: function(formula) {
-                try{
```
Comment (lasley) [inline web_widget_float_formula/static/src/js/web_widget_float_formula.js]: Why is this catch being removed? This was a bug fix for an edge case #431 abfea0fff77466e268dfad8dd007a45c3ffbc0c0

File: web_widget_float_formula/static/src/js/web_widget_float_formula.js [hunk 2]
```diff
@@ -85,12 +81,13 @@ odoo.define('web_widget_float_formula', function(require) {
...
                 this._clean_formula_text();
 
-                var formula = this._process_formula(this.$el.find('input').val());
+                var formula = this._process_formula(this.$el.val());
                 if (formula !== false) {
                     var value = this._eval_formula(formula);
                     if (value !== false) {
                         this._formula_text = "=" + formula;
                         this.set_value(value);
+                        this.$el.val(value);
```
Comment (lasley) [inline web_widget_float_formula/static/src/js/web_widget_float_formula.js]: This is circumventing the widget value set, which will not play nicely with inherited modules that may want to do other things with the value. Please remove - I'm almost certain nothing will break.

File: web_widget_float_formula/static/tests/js/test_web_widget_float_formula.js [hunk 3]
```diff
@@ -13,6 +13,7 @@ odoo.define_section('web_widget_float_formula', ['web.form_common', 'web.form_wi
         self.field = new form_widgets.FieldFloat(field_manager, filler);
         self.$element = $('<input>');
         self.field.$el.append(self.$element);
+        self.field.$label = $('<label>');
```
Comment (lasley) [inline web_widget_float_formula/static/tests/js/test_web_widget_float_formula.js]: Why was this needed?

File: web_widget_float_formula/static/tests/js/test_web_widget_float_formula.js [hunk 4]
```diff
@@ -116,10 +117,10 @@ odoo.define_section('web_widget_float_formula', ['web.form_common', 'web.form_wi
     test('._compute_result() should behave properly when the current value of the input element is a valid formula',
         function(assert, form_common, form_widgets, core) {
             window.test_setup(this, form_common, form_widgets, core);
-            this.$element.val('=2*3');
+            this.field.$el.val('=2*3');
```
Comment (lasley) [inline web_widget_float_formula/static/tests/js/test_web_widget_float_formula.js]: `this.$element` and `this.field.$el` are functionally equivalent, but the former is more explicit. Please switch back.

The reasoning behind this is that `this.field.$el` is a jQuery collection (array basically) that contains `this.$element`. jQuery operates on collections as a package, so `.val` is just setting all elements in the collection (`[this.$element]`) to the value.

**Review by lasley — CHANGES_REQUESTED** [review]
THanks @kellylougheed 

Looks like this doesn't work in products now though. Sales => Sales => Products => Any Product => Edit => Cost to `=2+3` => Save => Error on Cost

File: web_widget_float_formula/static/src/js/web_widget_float_formula.js (hunk 1 above)
Comment (kellylougheed) [reply web_widget_float_formula/static/src/js/web_widget_float_formula.js]: Oops, I must have accidentally deleted that with my console logs. Putting it back.

File: web_widget_float_formula/static/tests/js/test_web_widget_float_formula.js (hunk 3 above)
Comment (kellylougheed) [reply web_widget_float_formula/static/tests/js/test_web_widget_float_formula.js]: For the last two tests (which tested the change on blur and focus), this function was called: https://github.com/odoo/odoo/blob/6466aa499fc37bba5f4feeeecac99448ef88e62e/addons/web/static/src/js/views/form_common.js#L700 but _toggle_label failed because there was no this.$label in the test, so I put one in.

**Comment by kellylougheed** [comment]
Changes made @lasley

**Review by lasley — APPROVED** [review]

**Review by felixvillafranca — APPROVED** [review]
Great job ¡¡¡
